apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/hooks/useFeeBumping.ts

```diff
@@ -0,0 +1,265 @@
+import { useCallback } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { FeeBumpStrategy } from "@caravan/fees";
+import { FeeBumpStatus, FeeBumpRecommendation, FeePriority } from "../types";
+import { analyzeTransaction, extractUtxosForFeeBumping } from "../utils";
+import { updateBlockchainClient } from "../../../../../actions/clientActions";
+import { BlockchainClient } from "@caravan/clients";
+import {
+  useFeeBumpTransaction,
+  useFeeBumpTxHex,
+  useFeeBumpRecommendation,
+  useSelectedFeeBumpStrategy,
+  useSelectedFeeRate,
+  useSelectedFeePriority,
+  useFeeBumpResult,
+  useFeeBumpStatus,
+  useFeeBumpError,
+  useFeeBumpDispatch,
+  setFeeBumpTransaction,
+  setFeeBumpStatus,
+  setFeeBumpError,
+  setFeeBumpRecommendation,
+  setFeeBumpStrategy,
+  setFeeBumpRate,
+  setFeeBumpPriority,
+  resetFeeBumpState,
+} from "../context";
+
+/**
+ * Hook for handling transaction fee bumping with FeeBumping Redux State...
+ *
+ * This hook provides the core functionality for fee bumping in the Caravan wallet:
+ * 1. Analyzes transactions to determine if RBF or CPFP is possible
+ * 2. Gets real-time fee estimates from the blockchain using the `smart-fee` method
+ * 3. Creates fee-bumped transactions (RBF or CPFP)
+ *
+ * It only provides functions to perform operations that update the Redux state.
+ */
+export const useFeeBumping = () => {
+  const dispatch = useDispatch();
+  const feeBumpDispatch = useFeeBumpDispatch(); // dispatch for fee bump actions
+
+  // Get all state from Context
+  const transaction = useFeeBumpTransaction();
+  const txHex = useFeeBumpTxHex();
+  const status = useFeeBumpStatus();
+  const error = useFeeBumpError();
+  const recommendation = useFeeBumpRecommendation();
+  const selectedStrategy = useSelectedFeeBumpStrategy();
+  const selectedFeeRate = useSelectedFeeRate();
+  const selectedPriority = useSelectedFeePriority();
+  const result = useFeeBumpResult();
+
+  // Get wallet configuration from Redux store
+  const network = useSelector((state: any) => state.settings.network);
+  const addressType = useSelector((state: any) => state.settings.addressType);
+  const requiredSigners = useSelector(
+    (state: any) => state.settings.requiredSigners,
+  );
+  const totalSigners = useSelector((state: any) => state.settings.totalSigners);
```

Comment:
why not just: 

```suggestion
  const { network, addressType, requiredSigners, totalSigners } = useSelector((state: any) => state.settings));
```

you could even create a selector that is `selectWalletConfig` that would let you do `useSelector(selectWalletConfig)` (i think or something like that.